### ISSUE TITLE
Request to implement the API call StartDBInstanceAutomatedBackupsReplication

### ISSUE THREAD

**Issue by ttwd80** [hashicorp/terraform-provider-aws#20606]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

RDS instances support a feature called [Replicating automated backups to another AWS Region](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/USER_ReplicateBackups.html#AutomatedBackups.Replicating.Enable). If you have an RDS instance in `us-west-2`, you can configure the database to store the automated snapshots in a different region, for example `us-east-1`. The documentation shows that this can be done using AWS CLI [here](https://docs.aws.amazon.com/cli/latest/reference/rds/start-db-instance-automated-backups-replication.html). The AWS Go SDK API is [here](https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/rds#Client.StartDBInstanceAutomatedBackupsReplication).
Currently there is no way to configure a new RDS database to be created with automated backup replication to a different region (based on the documentation for the Terraform resource `aws_db_instance` [here](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/db_instance)).

### New or Affected Resource(s)

* aws_db_instance

### Potential Terraform Configuration

```hcl
resource "aws_db_instance" "default" {
  automated_backups_replication {
    backup_retention_period = 14
    target_region = "ap-south-1"
  }
}
```

### References

* https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/USER_ReplicateBackups.html#AutomatedBackups.Replicating.Enable

**Comment by davegallant** [comment]
I attempted to implement this in https://github.com/hashicorp/terraform-provider-aws/pull/23297. Any feedback or comments would be appreciated.

If anyone is looking for a more immediate solution, it may be worth checking out [aws_backup_vault](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/backup_vault).

**Comment by breathingdust** [comment]
Closed by #23759
